src/Compilers/Core/Portable/RealParser.cs

```diff
@@ -47,12 +56,20 @@ public static bool TryParseDouble(string s, out double d)
         /// <returns>True of the input was converted; false if there was an overflow</returns>
         public static bool TryParseFloat(string s, out float f)
         {
-            var str = DecimalFloatingPointString.FromSource(s);
-            var dbl = FloatFloatingPointType.Instance;
-            ulong result;
-            var status = RealParser.ConvertDecimalToFloatingPointBits(str, dbl, out result);
-            f = Int32BitsToFloat((uint)result);
-            return status != Status.Overflow;
+            try
+            {
+                var str = DecimalFloatingPointString.FromSource(s);
+                var dbl = FloatFloatingPointType.Instance;
+                ulong result;
+                var status = RealParser.ConvertDecimalToFloatingPointBits(str, dbl, out result);
+                f = Int32BitsToFloat((uint)result);
+                return status != Status.Overflow;
+            }
+            catch (System.FormatException)
```

Comment:
It occurs in `DecimalFloatingPointString.FromSource` on a string such as `"1.0e"` where the exponent is empty. See also related #6109.